src/test/java/seedu/address/logic/parser/schedule/EditEventCommandParserTest.java

```diff
@@ -0,0 +1,213 @@
+        Index targetIndex = INDEX_SECOND_PERSON;
+        String userInput = targetIndex.getOneBased() + " " + INDEX_FIRST_EVENT.getOneBased()
+                + EVENT_DESCRIPTION_DESC_CS2101 + DATE_DESC_PI_DAY + TIME_DESC_MORNING;
+
+        EditEventDescriptor descriptor = new EditEventDescriptorBuilder()
+                .withEventDescription(VALID_EVENT_DESCRIPTION)
+                .withDate(VALID_EVENT_DATE).withTime(VALID_EVENT_TIME).build();
+        EditEventCommand expectedCommand = new EditEventCommand(targetIndex, INDEX_FIRST_EVENT, descriptor);
+
+        assertParseSuccess(parser, userInput, expectedCommand);
+    }
+
+    // more test cases required
+    @Test
+    public void parse_oneFieldSpecified_success() {
+        Index targetIndex = INDEX_HENDRI;
+        Index targetEventIndex = INDEX_FIRST_EVENT;
+        // event description
+        String userInput = targetIndex.getOneBased() + " " + targetEventIndex.getOneBased()
+                + EVENT_DESCRIPTION_DESC_CS2101;
+        EditEventDescriptor descriptor = new EditEventDescriptorBuilder()
+                .withEventDescription(VALID_EVENT_DESCRIPTION).build();
+        EditEventCommand expectedCommand = new EditEventCommand(targetIndex, targetEventIndex, descriptor);
+        assertParseSuccess(parser, userInput, expectedCommand);
+
+        // date
+        userInput = targetIndex.getOneBased() + " " + targetEventIndex.getOneBased() + DATE_DESC_PI_DAY;
+        descriptor = new EditEventDescriptorBuilder().withDate(VALID_EVENT_DATE).build();
+        expectedCommand = new EditEventCommand(targetIndex, targetEventIndex, descriptor);
+        assertParseSuccess(parser, userInput, expectedCommand);
+
+        // time
+        userInput = targetIndex.getOneBased() + " " + targetEventIndex.getOneBased() + TIME_DESC_MORNING;
+        descriptor = new EditEventDescriptorBuilder().withTime(VALID_EVENT_TIME).build();
+        expectedCommand = new EditEventCommand(targetIndex, targetEventIndex, descriptor);
+        assertParseSuccess(parser, userInput, expectedCommand);
+
+        // duration
+        userInput = targetIndex.getOneBased() + " " + targetEventIndex.getOneBased() + DURATION_DESC_TWO_HOURS;
+        descriptor = new EditEventDescriptorBuilder().withDuration(VALID_EVENT_DURATION).build();
+        expectedCommand = new EditEventCommand(targetIndex, targetEventIndex, descriptor);
+        assertParseSuccess(parser, userInput, expectedCommand);
+
+        // recur frequency
+        userInput = targetIndex.getOneBased() + " " + targetEventIndex.getOneBased() + RECUR_FREQUENCY_DESC_WEEKLY;
+        descriptor = new EditEventDescriptorBuilder().withRecurFrequency(VALID_RECUR_FREQUENCY).build();
+        expectedCommand = new EditEventCommand(targetIndex, targetEventIndex, descriptor);
+        assertParseSuccess(parser, userInput, expectedCommand);
+    }
+
+    @Test
+    public void parse_multipleRepeatedFields_acceptsLast() {
```

Comment:
to follow the structure in AddCommandParserTest, can consider put this inside allFieldsSpecified_success(); or use parse_multipleRepeatedFields_success() as the name.